### PR TITLE
Add bazel 0.2.3

### PR DESCRIPTION
**Pull request by petemounce**
... because we need to peg the version so we can automate building CI workers while we update to support the later 0.3.1. We're happy to maintain past versions of bazel, here, since we're standardising on it as our build tool.

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-versions/pulls) for the same formula update/change?
  - I found #1306 (closed) which suggests that you might not want to accept this PR, but I wasn't sure - apologies if this is unwelcome in the repo.
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?
- [x] Does your submission adhere to Versions "Acceptable Formulae" [guidelines](https://github.com/Homebrew/homebrew-versions/blob/07b17dfb9f1/README.md#acceptable-formulae)?
  - I think so? Of the ones I think apply:
  - [x] am within +/- 2 major/minor versions
  - [x] conflicts_with bazel

### CODE REVIEW

File: bazel023.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+class Bazel023 < Formula
```
Comment (MikeMcQuaid) [inline bazel023.rb]: `Bazel02`

Comment (MikeMcQuaid) [inline bazel023.rb]: And name it `bazel02.rb`

File: bazel023.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  desc "Google's own build tool"
+  homepage "http://bazel.io/"
+  url "https://github.com/bazelbuild/bazel/archive/0.2.3.tar.gz"
+  sha256 "37fd2d49c57df171b704bf82c94e7bf954d94748e2a8621c5456c5c9d5f2c845"
+  head "https://github.com/bazelbuild/bazel.git"
+
+  conflicts_with "bazel", :because => "Differing version of same formula"
```
Comment (MikeMcQuaid) [inline bazel023.rb]: Move this below the `depends_on`: https://bot.brew.sh/job/Homebrew%20Versions%20Pull%20Requests/1561/version=yosemite/testReport/junit/brew-test-bot/yosemite/audit_homebrew_versions_bazel023___new_formula/

**Comment by MikeMcQuaid** [comment]
A few tweaks but I'm happy to accept this given that.



> We're happy to maintain past versions of bazel

Just FYI on Homebrew terminology: you cannot "maintain" these as you don't have commit access but if you can keep this up-to-date by submitting PRs that'd be great 👍

**Comment by petemounce** [comment]
@MikeMcQuaid thanks very much for speedy response.

**Comment by petemounce** [comment]
@MikeMcQuaid I'm also interested to pin the current version of bazel (which right now is 0.3.1) for the same reasons as this PR, so that when later versions are released our CI-agent-creation automation isn't silently broken. Is that a PR that would be accepted?

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for new formulae is `bazel02 0.2.3 (new formula)`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md). Thanks!

> Is that a PR that would be accepted?

Not in this tap yet, you're probably better to create your own tap for that for now. Please note we're planning on handling versions better in future: https://github.com/Homebrew/brew/issues/620.

**Comment by petemounce** [comment]
@MikeMcQuaid thanks

**Pull request closed without merging**